src/modules/SdnDiag.NetworkController/public/Get-SdnAuditLog.ps1

```diff
@@ -7,27 +7,33 @@ function Get-SdnAuditLog {
         Directory the results will be saved to. If ommitted, will default to the current working directory.
     .PARAMETER NcUri
         Specifies the Uniform Resource Identifier (URI) of the network controller that all Representational State Transfer (REST) clients use to connect to that controller.
-    .PARAMETER NCRestCredential
-        Specifies a user account that has permission to access the northbound NC API interface. The default is the current user.
+    .PARAMETER NcRestCertificate
+        Specifies the client certificate that is used for a secure web request to Network Controller REST API.
+        Enter a variable that contains a certificate or a command or expression that gets the certificate.
+    .PARAMETER NcRestCredential
+        Specifies a user account that has permission to perform this action against the Network Controller REST API. The default is the current user.
     .PARAMETER ComputerName
-        Type the NetBIOS name, an IP address, or a fully qualified domain name of one or more remote compute
+        Type the NetBIOS name, an IP address, or a fully qualified domain name of one or more remote computers.
     .PARAMETER Credential
-        Specifies a user account that has permission to perform this action. The default is the current user.
+        Specifies a user account that has permission to access the Computers. The default is the current user.
     #>
 
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'RestCredential')]
     param (
         [Parameter(Mandatory = $false)]
         [System.String]$OutputDirectory = "$(Get-WorkingDirectory)\AuditLogs",
 
         [Parameter(Mandatory = $true)]
         [Uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
         $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
 
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```

Comment:
Review `Mandatory` attribute

Reply:
This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.